metrics-api/src/main/java/software/amazon/swage/metrics/MetricContext.java

```diff
@@ -73,6 +72,25 @@ default void count(Metric label) {
         count(label, 1L);
     }
 
+    /**
+     * Create a child context with the supplied attributes.
+     * The child context will inherited the attributes from this instance and
+     * use the supplied attributes as overrides.
+     *
+     * @param attributes the attribute overrides to apply.
+     * @return a new child context with combined attributes.
+     */
+    MetricContext newChildContext(TypedMap attributes);
+
+    /**
+     * Create a new child context with attributes attributes inherited from this instance.
```

Comment:
What good is this method?

If the new child has the same attributes as the existing context, what makes it a different context?  Context are defined by the metadata comprising them (aka 'attributes'), doesn't make a lot of sense to have two logically distinct contexts which are functionally the same.